### ISSUE TITLE
Regression when trying to lazy load multiple relations when they are already loaded

### ISSUE THREAD

**Issue by smt116** [ash-project/ash#1434]
I've found a bug on 3.4.2. It fails when lazy loading more than one relation with `lazy?: true` when the source already includes them. i.e.,

```elixir
# This fails
Resource.by_id!("...", load: [:foo, :bar]) |> Ash.load!([:foo, :bar], lazy?: true)

# This works
Resource.by_id!("...") |> Ash.load!([:foo, :bar], lazy?: true)

# This works
Resource.by_id!("...", load: [:foo, :bar]) |> Ash.load!([:foo], lazy?: true) |> Ash.load!([:bar], lazy?: true)

# This works
Resource.by_id!("...", load: [:foo]) |> Ash.load!([:foo], lazy?: true)
```

```elixir
** (FunctionClauseError) no function clause matching in anonymous fn/2 in Ash.Actions.Read.Relationships.attach_related_records/2

    The following arguments were given to anonymous fn/2 in Ash.Actions.Read.Relationships.attach_related_records/2:

        # 1
        {%Ash.Resource.Relationships.BelongsTo{},
         {:lazy,
          #Ash.Query<>},
         %Task{}}

        # 2
        {:ok,
         [
           #Resource<>
         ]}

    (ash 3.4.2) lib/ash/actions/read/relationships.ex:29: anonymous fn/2 in Ash.Actions.Read.Relationships.attach_related_records/2
    (elixir 1.15.7) lib/enum.ex:4830: Enumerable.List.reduce/3
    (elixir 1.15.7) lib/enum.ex:2564: Enum.reduce_while/3
    (ash 3.4.2) lib/ash/actions/read/read.ex:293: Ash.Actions.Read.do_run/3
    (ash 3.4.2) lib/ash/actions/read/read.ex:81: anonymous fn/3 in Ash.Actions.Read.run/3
    (ash 3.4.2) lib/ash.ex:1841: Ash.load/3
    (ash 3.4.2) lib/ash.ex:1795: Ash.load/3
    (ash 3.4.2) lib/ash.ex:1736: Ash.load!/3
    iex:5: (file)
```

**Comment by zachdaniel** [comment]
Fixed in d144be692eb0fd691174713eb6100976d2d77bed

**Comment by zachdaniel** [comment]
Released as `3.4.3`
